### PR TITLE
chore(typescript in projet): add base for support typescript in yoga …

### PR DESCRIPTION
**Pull request by guilhermecardoso-gympass**
[JIRA Issue](https://gympass.atlassian.net/browse/)

## Description 📄

[Add Basic config for Yoga Support a typescript in project and configuration a doc and yoga package for suport a typescript]

## Platforms 📲

- [X] Web
- [ ] Mobile

## Type of change 🔍

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested? 🧪

[This change no need a test]

- [ ] Unit Test
- [ ] Snapshot Test

### CODE REVIEW

File: packages/doc/package.json [hunk 1]
```diff
@@ -14,6 +14,11 @@
     "@mdx-js/loader": "^1.4.5",
     "@mdx-js/mdx": "^1.4.5",
     "@mdx-js/react": "^1.4.5",
+    "@types/classnames": "^2.3.1",
+    "@types/node": "^18.11.17",
+    "@types/react": "^18.0.26",
+    "@types/react-dom": "^18.0.9",
+    "@types/react-helmet": "^6.1.6",
```
Comment (caiotracera) [inline packages/doc/package.json:21]: Shouldn't these `@types` and `typescript` be on `devDependencies` instead? They will be used only in development and should not be part of the final bundle

File: packages/yoga/package.json [hunk 2]
```diff
@@ -42,19 +43,26 @@
     "@gympass/yoga-tokens": "^3.1.5",
     "@ptomasroos/react-native-multi-slider": "^1.0.0",
     "@radix-ui/react-dropdown-menu": "^2.0.1",
+    "@types/jest": "^29.2.4",
+    "@types/react": "^18.0.26",
+    "@types/react-native": "^0.70.8",
+    "@types/react-test-renderer": "^18.0.0",
```
Comment (caiotracera) [inline packages/yoga/package.json:49]: Same here about if this should be in `devDependencies` instead here

**Review by caiotracera — COMMENTED** [review]
It's good to see that the typescript is finally coming to Yoga 🎉
I just have some comments regarding dependencies.

**Review by matheushdsbr — APPROVED** [review]

File: packages/yoga/.babelrc [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+  "extends": "../../babel.config.js",
+  "presets": [
+    [
+      "@babel/preset-typescript"
+    ]
+  ]
+}
```
Comment (MicaelRodrigues) [inline packages/yoga/.babelrc:8]: Don't forget the empty line at EOF, Here why:

https://gist.github.com/camh-/1bebfcff1b0f814e9b191edc60d5206b

**Pull request closed without merging**